source/plugin/data/custom/customdata.rst

```diff
@@ -0,0 +1,296 @@
+
+    import org.spongepowered.api.data.persistence.DataSerializable;
+    import org.spongepowered.api.data.persistence.DataContainer;
+    import org.spongepowered.api.data.persistence.DataQuery;
+
+    public class LastAttackerDataSerilizable implements DataSerializable {
+    
+        private UUID lastAttackerId;
+
+        public static final DataQuery UUID_PATH = DataQuery.of("attack", "last");
+
+        @Override
+        public int contentVersion(){
+            return 1;
+        }
+
+        @Override
+        public DataContainer toContainer(){
+            return DataContainer.createNew()
+                .set(LastAttackerDataSerilizable.UUID_PATH, lastAttackerId.toString());
+        }
+    
+    }    
+
+After that you will want to create a class that can build the data from a ``DataContainer`` this is known as 
+the :javadoc:`DataBuilder` which can be implemented as follows.
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.persistence.InvalidDataException;
+
+    public class LastAttackerDataBuilder implements DataBuilder<LastAttackerDataSerilizable> {
+    
+        @Override
+        public Optional<LastAttackerDataSerilizable> build(DataView container) throws InvalidDataException {
+            Optional<String> lastAttackerAsStringId container.getString(LastAttackerDataSerilizable.UUID_PATH);
+            if(lastAttackerAsStringId.isPresent()){
+                UUID lastAttacker = UUID.fromString(lastAttackerAsStringId.get());
+                return Optional.of(new LastAttackerDataSerilizable(lastAttacker));
+            }
+            return Optional.empty();
+        }
+    
+    }
+
+
+Registration
+============
+
+Registering your ``DataSerializable`` allows it to be accessible by Sponge and by other plugins in a generic way. The
+game/plugin can create copies of your data and serialize/deserialize your data without referencing any of your classes
+directly.
+
+To register a ``DataSerializable`` Sponge has the :javadoc:`RegisterDataEvent` event. This will allow you to register
+your data with the appropriate ``DataHolder``
+
+Registration Key
+================
+
+When it comes to registering your data, you are required to register it with a :javadoc:`Key` which will allow you and
+other developers access to your data manipulator.
+
+
+.. code-block:: java
+
+    import org.spongepowered.api.ResourceKey;
+    import org.spongepowered.api.data.Key;
+    import org.spongepowered.api.data.value.Value;
+
+    ResourceKey resourceKey = ResourceKey(pluginContainer, "last_attacker_manipulator");
+    Key<? extends Value<LastAttackerDataSerilizable>> key = Key
+        .builder()
+        .key(resourceKey)
+        .elementType(LastAttackerDataSerilizable.class)
+        .build();
+
+.. warning::
+
+    Be sure to store your ``Key`` somewhere global so you can access it later.
+
+.. tip::
+
+    You are able to register multiple keys for a single custom data for accessing specific parts of your data.
```

Comment:
This reads a bit fuzzy. Maybe it's just me reading "a single custom data" and thinking it's bad latin. 
But we're talking about making keys for some elements of a larger custom data set?

Reply:
Essentially yeah. If your wanting to modify a single element from a object, you can register a key for that element

Reply:
Rewritten